dropdown added, navbar fonts fixed, splash loading removed

RiceChrissy opened the pull request:
- [x] Selected and Unselected font size in navbar now the same

- [x] Removed circular loading indicator in splash screen

- [x] Added Functioning Dropdown Widget (Not yet designed properly)

![image](https://user-images.githubusercontent.com/72382027/116688529-087bd100-a9e1-11eb-9fd2-be18004457a2.png)


![image](https://user-images.githubusercontent.com/72382027/116688507-ff8aff80-a9e0-11eb-996a-043ee4a45587.png)

devnamme commented on lib/screens/qpi_page.dart:
```diff
@@ -1,15 +1,113 @@
+import 'package:admu_student_app/models/year.dart';
```
pls follow the imports format (so that our code is clean)

```dart
dart imports
// space
other package imports
// space
project imports
```
(also in alphabetical order)

devnamme commented on lib/screens/qpi_page.dart:
```diff
@@ -1,15 +1,113 @@
+
+            children:
+            _years.map((YearDropDown year){
+              return ExpansionPanel(
```
ideally, this `ExpansionPanel` should be in a separate file `lib/widgets/qpi/filename.dart`, and then import it in this file

it can look like this:
```dart
// in qpi_page.dart
// ...
Widget build(BuildContext context) {
  // ... other stuff like text, cumulative qpi, etc.
  ListBuilder
    // ...
    return YearDropdown();
```

tho i haven't tried the `ListView` widget yet, only `ListView.builder`

devnamme replied:
can you also briefly explain how this works?

also: i already have a draft of the QPI page in my branch (check `Emman/empty-screens-qpi` since i already linked it with the back-end

RiceChrissy replied:
Sorry for the very late reply, I made the dropdown component in a separate file now, as well as including the input fields and color pickers. 
To explain how the dropdown works, there's a class called YearDropDown that has 3 String variables, 1 header and 2 bodies.
![image](https://user-images.githubusercontent.com/72382027/116868732-85a18300-ac39-11eb-8ab4-7acc9ffa7f62.png)
I then made a list of YearDropDowns:
![image](https://user-images.githubusercontent.com/72382027/116868789-a2d65180-ac39-11eb-854b-87c0ac2a1357.png)
Then created an ExpansionPanelList Widget using the Strings got from YearDropDown.
Regarding your draft, what should I do with it to merge with the component I made?
If there's any way to improve this code please let me know.

devnamme replied:
oh i was referring to the `ListView` since i haven't used that one before; only `ListView.builder` so i dont know the differences between them.

supposedly, the QPI dropdown is the `ExpansionPanel` and not a list view:
```
// in qpi page

Column
  Text
  QPIView - cumulative qpi
  Text - qpi overview
  ListView.builder
    child:
      YearDropdown
```

something like that. think of one `widget` or `component` in the dev sprints as a singular button that you want to create over and over again across several files.
my goal here was that you guys in the front-end, won't have to copy and paste the same code for a single widget over and over and over again because that is tiring. and what if suddenly we have to change the border radius?

but don't worry, i can help out in creating the widgets for you guys ^^

regarding my draft, i'll be the one fixing the conflicts and hopefully by tomorrow, it's all in the develop branch and you guys can start working. thanks!

devnamme replied:
it's also the reason why i told the designers to switch to figma components

another example: the semester buttons. you see these buttons **at least** seven times in the add qpi page, and it's also included in the add schedule page.

dnamme reviewed (approved):
hmm, some of these are quite,,, odd but i'll fix it. goods for merging, expect that tomorrow everything will be done na!

RiceChrissy commented:
pushed some new code in this branch,

- [x] Added action buttons in the app bar
- [x] Added color picker & input field widgets
- [ ] Fixed dropdown component

Changed the way the app bar works so that it's different on every page; Basically, there are 4 app bars for each main screen
I made the drawer widget into its own file first

![image](https://user-images.githubusercontent.com/72382027/116953179-cee3e800-acb6-11eb-968b-9dc90d08e3d5.png)

![image](https://user-images.githubusercontent.com/72382027/116953327-26825380-acb7-11eb-8695-a356a4664148.png)


![image](https://user-images.githubusercontent.com/72382027/116953248-fc309600-acb6-11eb-8632-48e1cded6f52.png)

![image](https://user-images.githubusercontent.com/72382027/116953351-326e1580-acb7-11eb-9e6c-12138b3cd3d2.png)

devnamme commented on lib/widgets/select_semester.dart:
```diff
@@ -0,0 +1,49 @@
+
+import 'package:admu_student_app/widgets/select_color.dart';
+
+class SelectSemester extends StatefulWidget {
```
i assigned the select semester buttons to marco tho, i added that note in the notion page

RiceChrissy replied:
Yeah, I just made it because its code is similar to the SelectColor class, it can be deleted naman.

devnamme commented on lib/widgets/select_color.dart:
```diff
@@ -0,0 +1,124 @@
+import 'package:flutter/material.dart';
+
+class SelectColor extends StatefulWidget {
```
![image](https://user-images.githubusercontent.com/72055285/116955981-ae238e80-acc6-11eb-85ff-2c54bf815cea.png)
- overflow, is this the right size?

devnamme replied:
for the select color only

RiceChrissy replied:
I used the dimensions in figma (414x896) and they should fit perfectly, for phones with thinner width, do you want the color buttons to be smaller squares or thin rectangles?

devnamme replied:
not sure to be honest but it might look better if we use smaller squares.

let's fix stuff like this probably next week when we have the complete qpi screens

devnamme commented on lib/widgets/whole_text_field.dart:
```diff
@@ -0,0 +1,32 @@
+import 'package:flutter/material.dart';
+
+class WholeTextField extends StatefulWidget {
```
i tested it and the `TextField` class automatically expands (width) according to its parent

![image](https://user-images.githubusercontent.com/72055285/116956145-170b0680-acc7-11eb-9eeb-137dab149ab2.png)

to reduce code, you can simply merge all three classes into one:

```
class ??? extends StatefulWidget
  accept a boolean if it's multilined or not
  accept an integer for maxlength
}

class ??? state
  return Container() -- for box decoration with child of TextInput
  -- never add padding for singular widgets, let the page code handle that
```

RiceChrissy replied:
Alright I'll try this

devnamme replied:
alright, that should be the last commit or test for this branch. i'll fix the conflicts asap (after your commit) so that we can merge it to develop!

please don't create a new branch before i merge it

RiceChrissy replied:
Done! Thanks for the advice :>

![image](https://user-images.githubusercontent.com/72382027/116964668-4ecd7a80-acd6-11eb-92c7-320b3edd1c9c.png)

RiceChrissy replied:
When this is merged I'll pull from develop and work on the other widgets

dnamme reviewed (changes requested):
merge text fields into a single class
i also don't know why you don't have the onboarding+login pages but that's alright

dnamme reviewed (approved):
i'll merge this one later, you can start on this week's tasks probably tomorrow morning or hopefully by midnight tonight! don't create a new branch from develop until then. great job dude!

dnamme commented:
![image](https://user-images.githubusercontent.com/72055285/116984506-e6928f00-acfd-11eb-95b0-527c0421df52.png)

hello, I'm afraid that I have to undo your changes with the appbar since it's not behaving properly (drawer should cover the bottom nav bar). dw i'll be the one fixing it
